packages/commerce-sdk-react/src/provider.tsx

```diff
@@ -42,11 +43,21 @@ export const CommerceApiContext = React.createContext({} as ApiClients)
  * @returns Provider to wrap your app with
  */
 const CommerceApiProvider = (props: CommerceApiProviderProps): ReactElement => {
-    const {children, clientId, organizationId, shortCode, siteId, proxy, queryClientConfig} = props
+    const {
+        children,
+        clientId,
+        correlationId = '',
```

Comment:
@kevinxh What are your thoughts on promoting the correlation id header to be its own prop? Since you are the key stakeholder of the commerce-sdk-react I want to make sure this is following your vision.

Reply:
from the `commerce-sdk-react` library's perspective, this is just an HTTP header, nothing special that make it stand out from other headers. So I'd suggest we make the provider accept a `headers` prop.